src/ship_cmd.cpp

```diff
@@ -328,11 +328,11 @@ void Ship::UpdateDeltaXY()
 }
 
 /**
- * Test-procedure for HasVehicleOnPos to check for a ship.
+ * Test-procedure for HasVehicleOnPos to check for any ships which are visible and not stopped by the player.
  */
-static Vehicle *EnsureNoVisibleShipProc(Vehicle *v, void *data)
+static Vehicle *EnsureNoActiveShipProc(Vehicle *v, void *data)
 {
-	return v->type == VEH_SHIP && (v->vehstatus & VS_HIDDEN) == 0 ? v : nullptr;
+	return (v->type == VEH_SHIP && ((v->vehstatus & VS_HIDDEN) || (v->vehstatus & VS_STOPPED))) == 0 ? v : nullptr;
```

Comment:
This test doesn't do what you think it does :)
```suggestion
	return v->type == VEH_SHIP && (v->vehstatus & (VS_HIDDEN | VS_STOPPED)) == 0 ? v : nullptr;
```

Reply:
Thanks. I haven't quite wrapped my head around bitwise operation yet.